### PR TITLE
When clamping surrogate loss, do this based on absolute value

### LINKED ISSUE #37334: [RLlib] PPO surrogate loss incorrectly clipped
### What happened + What you expected to happen

In both the torch and TF implementations the surrogate loss calculation is incorrectly clipped. The `clip_param` should normally clip the `logp_ratio` (by default between 0.7 and 1.3). However when you look at the torch implementation:

```
surrogate_loss = torch.min(
            train_batch[Postprocessing.ADVANTAGES] * logp_ratio,
            train_batch[Postprocessing.ADVANTAGES]
            * torch.clamp(
                logp_ratio, 1 - self.config["clip_param"], 1 + self.config["clip_param"]
            ),
        )
```

You can see that the `torch.min` is taken. In case of negative advantages, a big positive `logp_ratio` in combination with a negative advantage, this will result an a huge negative `surrogate_loss`. In my environment this could amount to even billions of negative surrogate loss.

A proposed fix for torch, but this should also be fixed for tensorflow.

```
absolute_advantages = torch.abs(train_batch[Postprocessing.ADVANTAGES])
        clamped_logp_ratio = torch.clamp(
            logp_ratio, 1 - self.config["clip_param"], 1 + self.config["clip_param"])
        surrogate_loss = torch.sign(train_batch[Postprocessing.ADVANTAGES]) * torch.min(
            absolute_advantages * logp_ratio,
            absolute_advantages * clamped_logp_ratio)
```

### Versions / Dependencies

RlLib 3.0.0

### Reproduction script

Not needed

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by MattiasDC**
## Why are these changes needed?
Fixes #37334. Surrogate loss is now clamped based on the absolute value, i.s.o. on the value, which could result in huge negative surrogate loss values to be preferred, neglecting clamping.

## Related issue number
Closes #37334 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ArturNiederfahrenhorst** [comment]
@MattiasDC Many thanks for this PR! 

To me it looks like our loos correctly implements the clipped objective from the original paper:
Screenshot 2023-07-18 at 16 10 04
Furthermore, why would we first take the absolute advantage and then multiply the whole term by the sign later?

Assuming that your claims are valid, these are the changes that we would still need in this PR: 

We are currently doing a migration from Policy/ModelV2 to RLModules and Learners.
One can find these inside the algorithms/ppo directory where you also found the policy.
The code that this PR currently touches are not used by RLlib anymore by default.
If we would not keep them the same, the implementations would diverge (we made sure Policy/ModelV2 do the same thing as RL
Modules for now).

**Comment by MattiasDC** [comment]
Hi @ArturNiederfahrenhorst,

The algorithm is indeed implemented as the paper describes.

The absolute was taken to first determine the minimum of the absolute, and then reapply the sign. This is different from taking the minimum of the value, e.g. in my PR with advantages -10 and -2, my PR would take -2, whilst the current would take -10, if the logp_ratio would be 1. I do still have the issue of exploding probability ratio (in my case it exploded to something in the range of 10^6), which completely dominates the whole loss, therefor I though the clip_param was to prevent this, but this is only intended to clip the ratio, in one direction.
In my case the advantages are negative and the ratio is very large, causing no clipping, but resulting in something like -2*10^6 = -20^6 surrogate loss, whereas my implementation would use the clipped the ratio.

I'll close this PR and find the reason as to why my ratio is exploding.
